presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/RowEncoder.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder;
+
+import io.prestosql.spi.block.Block;
+
+public interface RowEncoder
+{
+    /**
+     * Adds the value from the given block/position to the row being encoded
+     */
+    RowEncoder appendColumnValue(Block block, int position);
+
+    /**
+     * Returns the encoded values as a byte array
+     */
+    byte[] toByteArray();
+
+    /**
+     * Resets the encoder so that it can be used for the next row
+     */
+    void reset();
```

Comment:
I wonder if we need this method at all.  Perhaps the contract could be: add data, then generate the record, rinse, repeat.  Generating the `byte[]` could reset things (that would be an implementation detail).

Reply:
I was thinking a similar thing. I just went through and updated all the encoders with the changes I've made and I realized that the `clear` method  was only used once by the `RawRowEncoder`.